CinemaTicket.BusinessLogicServices/MovieService.cs

```diff
@@ -101,7 +111,14 @@ public async Task CreateAsync(MovieCreate movieCreate)
             }
             if (needAddGenres != null && needAddGenres.Count > 0)
             {
-                var genres = needAddGenres.Select(x => new Genre
```

Comment:
var genresLower = movieCreate.GenreNames.Select(x => x.ToLower()).ToList();
            var genresFromDb = await genreDataAccess.GetGenreListAsync(genresLower);
            var newGenres = new List<Genre>();
            foreach (var genre in movieCreate.GenreNames)
            {
                var genreExist = genresFromDb.FirstOrDefault(x => x.Name.ToLower() == genre.ToLower());
                if (genreExist == null)
                {
                    movie.Genres.Add(genreExist);
                }
                else
                {
                    newGenres.Add(new Genre
                    {
                        Name = genre,
                        CreatedOn = DateTime.Now,
                        ModifiedOn = DateTime.Now,
                        CreatedBy = currentUser.Id,
                        ModifiedBy = currentUser.Id,
                        Movies = new List<Movie> { movie }
                    });
                }
            }

            await genreDataAccess.CreateListAsync(newGenres);
            await movieDataAccess.CommitAsync();